docs/cheat-sheets/java-xxe.mdx

```diff
@@ -0,0 +1,1839 @@
+---
+slug: java-xxe
+toc_max_heading_level: 4
+---
+
+# XML External Entity prevention for Java
+
+This is an XML External Entity (XXE) prevention cheat sheet by r2c. It contains code patterns of potential XXE in an application. The recommendations in this cheat sheet pave a safe road for developers that mitigate the possibility of XXE in their code. By following these recommendations, you can be reasonably sure that their code is free of XXE.
+
+## Mitigation Summary
+
+It is not easy to summarize Java XML security. The XML standard allows multiple ways to include external content, including but not limited to external Document Type Definitions (DTD), external Entity References to external data, general entity references, and external parameter references. There are also several ways to include external content in XML Schemas and stylesheets.
+
+The Java API for XML Processing (JAXP) contains multiple interfaces to process XML content:
+
+- The Document Object Model (DOM) interface
+- The Simple API for XML (SAX) interface
+- The Streaming API for XML (StAX) interface
+
+Each of these interfaces has its mitigation measures to disable the use of different sources of external content. The&nbsp;inconsistency in availability and even the effect of these measures across the different APIs, combined with the variety of ways to include external content makes it difficult to provide general mitigation recommendations. Many configurations that are secure for one parser are not secure for another.
```

Comment:
Why the `&nbsp;` in `The&nbsp;inconsistency`?

Reply:
Just to prevent a line break after The. But I guess this does not make sense as everyone will have a different resolution.

Reply:
```suggestion
Each of these interfaces has its mitigation measures to disable the use of different sources of external content. The inconsistency in availability and even the effect of these measures across the different APIs, combined with the variety of ways to include external content makes it difficult to provide general mitigation recommendations. Many configurations that are secure for one parser are not secure for another.
```